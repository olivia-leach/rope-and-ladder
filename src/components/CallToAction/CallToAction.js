import React, { Component } from 'react';
import "../../styles/Email.scss";

import './CallToAction.scss'
import Footer from '../Footer';
import { Element } from 'react-scroll';

class CallToAction extends Component {
    render() {
        return (
            <Element name='listen' className='page-section sub-section'>
        <div className='call-to-action-container'>
            <div className='iframe-container'>
                <iframe src="https://open.spotify.com/embed/track/45q74mP57nDkU7Wjetk3ak" width="400" height="480" frameBorder="0" allowtransparency="true" allow="encrypted-media"></iframe>
            </div>
            <div className='flex flex-center input-container'>
                <a className="mdc-button mdc-button--raised" href='https://www.smarturl.it/the-conspiracy' target='_blank'>
                    <span className="mdc-button__label">stream the conspiracy</span>
                </a>
            </div>
        </div>
        </Element>
        )
    }
}

export default CallToAction;
