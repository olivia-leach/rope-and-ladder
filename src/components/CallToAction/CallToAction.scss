$break-small: 767px;

.call-to-action-container {
    min-height: 62vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 0 1.3rem;

    .input-container {
        margin-top: 2rem;
    }
  }

.call-to-action-instructions {
    font-size: 0.98rem;
    font-weight: 400;
    letter-spacing: 2px;
    margin: 2rem 0;
    color: #AAAAAA;
    text-transform: uppercase;
}

.iframe-container {
    display: flex;
    justify-content: center;

    @media screen and (max-width: $break-small) {
        iframe {
            width: 300px;
            height: 380px;
        }

        flex-direction: column;
        align-items: center;
      }
}

.link-container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin: 1px 1px 1px 1rem;

    @media screen and (max-width: $break-small) {
        margin: 1rem 0 0;

        .button-link {
            margin: 0.4rem 0;
        }
    }
}