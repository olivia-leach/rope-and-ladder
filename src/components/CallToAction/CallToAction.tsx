import React from 'react'
import { Element } from 'react-scroll'

import './CallToAction.scss'
import ButtonLink from '../ButtonLink/ButtonLink'

const CallToAction = () => (
  <Element name="listen" className="page-section">
    <div className="call-to-action-container">
      <div className="iframe-container">
        <iframe
          title="play rope and ladder music"
          src="https://open.spotify.com/embed/album/3V6cYBEdgKndJBXxuxqU5o"
          width="400"
          height="480"
          frameBorder="0"
          allowTransparency
          allow="encrypted-media"
        />
        <div className="link-container">
          <ButtonLink
            suppressNewPage
            filled
            link="/#store"
            label="official store"
            icon="fa fa-shopping-cart"
          />
          <ButtonLink
            filled
            link="https://www.smarturl.it/rope-and-ladder/spotify"
            label="spotify"
            icon="fab fa-spotify"
          />
          <ButtonLink
            filled
            link="https://www.smarturl.it/rope-and-ladder/youtubemusic"
            label="youtube"
            icon="fab fa-youtube-square"
          />
          <ButtonLink
            filled
            link="https://www.smarturl.it/rope-and-ladder/bandcampbuy"
            label="bandcamp"
            icon="fab fa-bandcamp"
          />
          <ButtonLink
            filled
            link="https://www.smarturl.it/rope-and-ladder/applemusic"
            label="apple music"
            icon="fab fa-apple"
          />
          <ButtonLink
            filled
            link="https://www.smarturl.it/rope-and-ladder/itunes"
            label="itunes"
            icon="fab fa-itunes-note"
          />
          <ButtonLink
            filled
            link="https://www.smarturl.it/rope-and-ladder/az"
            label="amazon"
            icon="fab fa-amazon"
          />
          <ButtonLink
            filled
            link="https://www.smarturl.it/rope-and-ladder/googleplay"
            label="google play"
            icon="fab fa-google-play"
          />
        </div>
      </div>
    </div>
  </Element>
)

export default CallToAction
