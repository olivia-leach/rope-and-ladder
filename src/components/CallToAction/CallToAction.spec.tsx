import { render } from '@testing-library/react'
import React from 'react'
import CallToAction from './CallToAction'

describe('CallToAction', () => {
  it('should match snapshot', () => {
    const { container } = render(<CallToAction />)
    expect(container).toMatchSnapshot()
  })
})
