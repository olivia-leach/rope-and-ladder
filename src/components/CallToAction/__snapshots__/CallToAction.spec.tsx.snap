// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction should match snapshot 1`] = `
<div>
  <div
    class="page-section"
    name="listen"
  >
    <div
      class="call-to-action-container"
    >
      <div
        class="iframe-container"
      >
        <iframe
          allow="encrypted-media"
          frameborder="0"
          height="480"
          src="https://open.spotify.com/embed/album/3V6cYBEdgKndJBXxuxqU5o"
          title="play rope and ladder music"
          width="400"
        />
        <div
          class="link-container"
        >
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="/#store"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fa fa-shopping-cart"
                />
                 
                official store
              </span>
            </div>
          </a>
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="https://www.smarturl.it/rope-and-ladder/spotify"
            target="_blank"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fab fa-spotify"
                />
                 
                spotify
              </span>
            </div>
          </a>
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="https://www.smarturl.it/rope-and-ladder/youtubemusic"
            target="_blank"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fab fa-youtube-square"
                />
                 
                youtube
              </span>
            </div>
          </a>
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="https://www.smarturl.it/rope-and-ladder/bandcampbuy"
            target="_blank"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fab fa-bandcamp"
                />
                 
                bandcamp
              </span>
            </div>
          </a>
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="https://www.smarturl.it/rope-and-ladder/applemusic"
            target="_blank"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fab fa-apple"
                />
                 
                apple music
              </span>
            </div>
          </a>
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="https://www.smarturl.it/rope-and-ladder/itunes"
            target="_blank"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fab fa-itunes-note"
                />
                 
                itunes
              </span>
            </div>
          </a>
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="https://www.smarturl.it/rope-and-ladder/az"
            target="_blank"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fab fa-amazon"
                />
                 
                amazon
              </span>
            </div>
          </a>
          <a
            class="button-link button-link--filled"
            data-testid="buttonLink"
            href="https://www.smarturl.it/rope-and-ladder/googleplay"
            target="_blank"
          >
            <div
              class="button-link__div"
            >
              <span
                class="button-link__span"
              >
                <i
                  class="fab fa-google-play"
                />
                 
                google play
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
